command_line_tools.md

```diff
@@ -0,0 +1,71 @@
+# Command Line Tools
+
+When writing command line tools, there are a couple of pitfalls to be aware of.
+In the past we made certain painful experiences which we would like to prevent
+in the future.
+
+### Project Structure
+
+Let's consider an example project for our new command line tool called
+`cooltool`. The project structure should look something like this.
+
+```
+$ tree cooltool -d
+cooltool
+└── cmd
+    ├── create
+    │   └── cluster
+    │       ├── defaulting
```

Comment:
Is it sensible to add code under specific command instead of sharing it via proper packages in top level? I have noticed we have this already in `gsctl`, but I'm afraid this leads to code being copied instead of making it more generic for sharing.

Code used for defaulting and validation is often something that is probably useful for multiple different commands so I think this even emphasizes the question here.

Reply:
I think the correct answer here is "depends". When there are cases in which we want to reuse certain rules we can think about this. This here is basically the case of iterating early on and revisit it later again. We just have to start with something and I think the suggestion is reasonable for a starter.